packages/@aws-cdk/aws-eks-v2-alpha/lib/kubectl-handler/helm/__init__.py

```diff
@@ -105,42 +105,59 @@ def get_oci_cmd(repository, version):
+    helm_cmd.append('--untar')
+
     if private_registry['registry'] is not None:
         logger.info("Found AWS private repository")
-        cmnd = [
-            f"aws ecr get-login-password --region {private_registry['region']} | " \
-            f"helm registry login --username AWS --password-stdin {private_registry['registry']}; helm pull {repository} --version {version} --untar"
-            ]
+        login_cmd = ['aws', 'ecr', 'get-login-password', '--region', private_registry['region']]
+        registry_login = ['helm', 'registry', 'login', '--username', 'AWS', '--password-stdin', private_registry['registry']]
+        return {'login': login_cmd, 'registry_login': registry_login, 'helm': helm_cmd}
     elif public_registry['registry'] is not None:
         logger.info("Found AWS public repository, will use default region as deployment")
         region = os.environ.get('AWS_REGION', 'us-east-1')
 
         if is_ecr_public_available(region):
-            cmnd = [
-                f"aws ecr-public get-login-password --region us-east-1 | " \
-                f"helm registry login --username AWS --password-stdin {public_registry['registry']}; helm pull {repository} --version {version} --untar"
-                ]
+            login_cmd = ['aws', 'ecr-public', 'get-login-password', '--region', 'us-east-1']
+            registry_login = ['helm', 'registry', 'login', '--username', 'AWS', '--password-stdin', public_registry['registry']]
+            return {'login': login_cmd, 'registry_login': registry_login, 'helm': helm_cmd}
         else:
-            # `aws ecr-public get-login-password` and `helm registry login` not required as ecr public is not available in current region
+            # No login required for public ECR in non-aws regions
             # see https://helm.sh/docs/helm/helm_registry_login/
-            cmnd = [f"helm pull {repository} --version {version} --untar"]
+            return {'helm': helm_cmd}
     else:
         logger.error("OCI repository format not recognized, falling back to helm pull")
-        cmnd = [f"helm pull {repository} --version {version} --untar"]
-
-    return cmnd
+        return {'helm': helm_cmd}
 
 
 def get_chart_from_oci(tmpdir, repository = None, version = None):
-
-    cmnd = get_oci_cmd(repository, version)
+    from subprocess import Popen, PIPE
+    
+    commands = get_oci_cmd(repository, version)
 
     maxAttempts = 3
     retry = maxAttempts
     while retry > 0:
         try:
-            logger.info(cmnd)
-            output = subprocess.check_output(cmnd, stderr=subprocess.STDOUT, cwd=tmpdir, shell=True)
+            # Execute login commands if needed
+            if 'login' in commands:
+                logger.info("Running login command: %s", commands['login'])
+                
+                # Create pipeline: aws ecr get-login-password | helm registry login
+                p1 = Popen(commands['login'], stdout=PIPE, stderr=subprocess.STDOUT, cwd=tmpdir)
```

Comment:
Any error should be passed on as standard error, not standard output.

```suggestion
                p1 = Popen(commands['login'], stdout=PIPE, cwd=tmpdir)
```

Reply:
A few lines below we check for "Broken pipe" in the final output. For simplicity, I'm capturing both err and out in the same into one so the check is simpler. By default, `Popen` will not throw an error when failing, but returns `stdout, stderr` tuple when `.communicate()` is called on the subprocess